### ISSUE TITLE
CA1861 false positive in method attribute

### ISSUE THREAD

**Issue by koenigst** [dotnet/roslyn#70825]
**Version Used**: 
.NET Core SDK 8.0.100

**Steps to Reproduce**:
1. Enable all analyzers.
```xml
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisMode>All</AnalysisMode>
```

2. Add attribute with array.
```csharp
public class Test
{
    [Xunit.Theory]
    [Xunit.InlineData("a", new[] { "b" })]
    public void Method(string a, string[] b) { }
}
```

**Diagnostic Id**:
CA1861

**Expected Behavior**:
The array `new[] { "b" }` has no compiler warnings.

**Actual Behavior**:
The array `new[] { "b" }` has a CA1861 warning but the code cannot be changed because no field is allowed in attribute syntax.
